src/analyses/call_graph.cpp

```diff
@@ -51,50 +46,158 @@ void call_grapht::add(
   const irep_idt &caller,
   const irep_idt &callee)
 {
-  graph.insert(std::pair<irep_idt, irep_idt>(caller, callee));
+  std::size_t caller_idx = node_numbering.number(caller);
+  if(caller_idx >= nodes.size())
+  {
+    node_indext node_index = add_node();
+    nodes[node_index].function_name = caller;
+  }
+
+  std::size_t callee_idx = node_numbering.number(callee);
+  if(callee_idx >= nodes.size())
+  {
+    node_indext node_index = add_node();
+    nodes[node_index].function_name = callee;
+  }
+
+  add_edge(caller_idx, callee_idx);
+}
+
+
+void call_grapht::output_dot_node(std::ostream &out, node_indext n) const
+{
+  const nodet &node = nodes.at(n);
+
+  for(const auto &edge : node.out)
+  {
+    out << "  \"" << node.function_name << "\" -> " << "\""
+        << nodes[edge.first].function_name << "\" " << " [arrowhead=\"vee\"];"
+        << "\n";
+  }
+}
+
+void call_grapht::output_xml_node(std::ostream &out, node_indext n) const
+{
+  const nodet &node = nodes.at(n);
+
+  for(const auto &edge : node.out)
+  {
+    out << "<call_graph_edge caller=\"";
+    xmlt::escape_attribute(id2string(node.function_name), out);
+    out << "\" callee=\"";
+    xmlt::escape_attribute(id2string(nodes[edge.first].function_name), out);
+    out << "\">\n";
+  }
+}
+
+void call_grapht::output_xml(std::ostream &out) const
+{
+  for(node_indext n = 0; n < nodes.size(); n++)
+    output_xml_node(out, n);
 }
```

Comment:
Would asking for an output_json be unreasonable?

Reply:
Not unreasonable, it should also be added to grapht, which doesn't have one either.

Reply:
That'd be lovely.